app/routes/__index/domains/index.tsx

```diff
@@ -1,22 +1,38 @@
 import { AddIcon } from '@chakra-ui/icons';
 import { Button, Container, Flex, Heading, Text } from '@chakra-ui/react';
+import type { LoaderArgs } from '@remix-run/node';
+import { Response } from '@remix-run/node';
 import { Link, useNavigate } from '@remix-run/react';
 import type { DomainsTableAction } from '~/components/domains-table';
 import DomainsTable from '~/components/domains-table';
-import DOMAINS_MOCK from '~/mocks/domains';
+import { getRecordsByUsername } from '~/models/record.server';
+import { getUser } from '~/session.server';
+import type { Record } from '@prisma/client';
+import { typedjson, useTypedLoaderData } from 'remix-typedjson';
```

Comment:
nit:  Separate your imports such that you do third-party, internal, types:

```ts
import foo from 'foo';
...

import { getUser } from '~/session.server';
...

import type { Record } from '@prisma/client';
...
```

This makes it a lot easier to understand what's happening.